pkg/portal/ssh/proxy.go

```diff
@@ -281,8 +281,8 @@ func (s *ssh) newChannel(ctx context.Context, accessLog *logrus.Entry, nc crypto
 	defer recover.Panic(s.log)
 
 	ch2, rs2, err := downstreamConn.OpenChannel(nc.ChannelType(), nc.ExtraData())
-	if err, ok := err.(*cryptossh.OpenChannelError); ok {
-		return nc.Reject(err.Reason, err.Message)
+	if errAsOpenChannel, ok := err.(*cryptossh.OpenChannelError); ok {
+		return nc.Reject(errAsOpenChannel.Reason, errAsOpenChannel.Message)
 	} else if err != nil {
```

Comment:
this breaks the control flow in the following else if

Reply:
it does not. if anything, what was there was confusing at best. The `err` in the `else if` is the one from before the whole `if` statement. I originally had that in the PR because I also had a golangci-lint update and the new version was complaining about it (this check specifically https://staticcheck.io/docs/checks#SA9008) but I'll remove it

Reply:
ope totally see what you mean now. this definitely is a bug fix then as the `else if` was using the closer scoped `err` as type `cryptossh.OpenChannelError` when the `else if` should've been referencing the `err` of type `error`.

Reply:
even typing that I confused myself ugh

Reply:
not only that, but also what we would get is `nil` because when we do a type assertion and `ok == false` it means that `err` is the zero value of `*openChannelError` (`nil`), so the `else if` is never true 

I would say that it's generally a bad idea to have variable overrides like this one. It's very rarely useful and quite often very confusing